hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/backend/AbstractBackendEntityEnricher.java

```diff
@@ -122,6 +122,21 @@ protected boolean canResolveBackend(StructuredTraceGraph structuredTraceGraph, E
     return true;
   }
 
+  /**
+   * Method to check if backend uri should be processed or not. This will enable any custom logic to
+   * be inserted in the implementing classes.
+   *
+   * @param structuredTraceGraph structured trace graph
+   * @param event leaf exit span
+   * @param backendURI backend URI information
+   * @return true if backend uri resolution is allowed
+   */
+  protected boolean shouldResolveBackendUri(
```

Comment:
Can we just make it part of `canResolveBackend`, and get rid of the other method?